File: src/contests/cg/puzzle/python3/distributing-candy.py
# 🎮 CodinGame Puzzle - distributing-candy
# https://www.codingame.com/training/easy/distributing-candy

n, m = map(int, input().split())
candies = sorted(map(int, input().split()))
best = float("inf")
for i in range(0, n - m + 1):
    diff = candies[i + m - 1] - candies[i]
    if diff < best:
        best = diff
print(best)

File: src/contests/cg/puzzle/scala/distributing-candy.scala
// 🎮 CodinGame Puzzle - distributing-candy
// https://www.codingame.com/training/easy/distributing-candy

object Solution {
  def main(args: Array[String]): Unit = {
    val Array(n, m) = scala.io.StdIn.readLine().trim.split(" ").map(_.toInt)
    val candies = scala.io.StdIn.readLine().trim.split(" ").map(_.toInt).sorted
    var best = Int.MaxValue
    var i = 0
    while (i + m - 1 < n) {
      val diff = candies(i + m - 1) - candies(i)
      if (diff < best) best = diff
      i += 1
    }
    println(best)
  }
}
